src/panels/calendar/dialog-calendar-event-detail.ts

```diff
@@ -122,16 +128,69 @@ class DialogCalendarEventDetail extends LitElement {
   }
 
   private _renderRruleAsText(value: string) {
-    // TODO: Make sure this handles translations
     try {
-      const readableText =
-        value === "" ? "" : RRule.fromString(`RRULE:${value}`).toText();
-      return html`<div id="text">${readableText}</div>`;
+      const rule = RRule.fromString(`RRULE:${value}`);
+      if (rule.isFullyConvertibleToText()) {
+        const readableText =
+          value === ""
+            ? ""
+            : RRule.fromString(`RRULE:${value}`).toText(
+                this._translateRruleElement.bind(this),
+                {
+                  dayNames: this._getDayNames(this.hass.locale),
+                  monthNames: this._getMonthNames(this.hass.locale),
+                  tokens: {},
+                },
+                this._formatDate.bind(this)
+              );
+        return html`<div id="text">${readableText}</div>`;
+      }
+
+      return html`<div id="text">Cannot convert recurrence rule</div>`;
     } catch (e) {
+      return "Error while processing the rule";
+    }
+  }
+
+  private _translateRruleElement(id: string | number | Weekday): string {
+    if (typeof id === "string") {
+      return this.hass.localize(`ui.components.calendar.event.rrule.${id}`);
+    }
+
+    return "";
+  }
+
+  private _formatDate(year: number, month: string, day: number): string {
+    if (!year || !month || !day) {
       return "";
     }
+
+    // Build date so we can then format it
+    const date = new Date();
+    date.setFullYear(year);
+    // As input we already get the localized month name, so we now unfortuantely
```

Comment:
Why do we provide translations for them then? If we keep them in english it will be easier?

Reply:
I see no way to do that. `monthNames` has to be provided. If I provide an empty `string[]` we get empty months in `_formatDate()`.

I think this double conversion is a limitation in the `rrule` library.